addons/payment/migrations/13.0.1.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,47 @@
+---Models in module 'payment'---
+new model payment.link.wizard [transient]
+# NOTHING TO DO
+
+---Fields in module 'payment'---
+payment      / account.invoice          / transaction_ids (many2many)   : DEL relation: payment.transaction
+payment      / account.move             / transaction_ids (many2many)   : NEW relation: payment.transaction
+# NOTHING TO DO: many2many table didn't changed
```

Comment:
but you should update the invoice_id values of this account_invoice_transaction_rel table.

Reply:
why? and how should i do that?

Reply:
Because the invoice_id points to a different place. Before it was pointing to some id in account_invoice and now should point to some id in account_move, and thus probably not the same id.

add in post-migration
```python
def update_account_invoice_transaction_rel_table(env):
    openupgrade.logged_query(
        env.cr, """
        UPDATE account_invoice_transaction_rel aitr
        SET invoice_id = am.id
        FROM account_move am
        WHERE am.old_invoice_id = aitr.old_invoice_id""",
    )
```

obviously, this only makes sense if my PR of account module gets merged as it is now. that's why we, the people here in this repo, try to not make many PRs in advance, because some dependencies become key to proceed and move forward 😉

Reply:
Is this because that table was merged and ids changed? 
Do I have to copy in old_invoice_id in account_invoice_transaction_rel and account_move tables in pre-migration first?

Reply:
yes. and no need.